ats/bin/atsflux.py

```diff
@@ -0,0 +1,133 @@
+        help="Exclusive use of nodes allocated.",
+    )
+    parser.add_argument(
+        "-A",
+        "--account",
+        dest="account",
+        default="guests",
+        type=str,
+        help="Project bank to charge.",
+    )
+    parser.add_argument(
+        "-p",
+        "--partition",
+        default="pdebug",
+        type=str,
+        help="Partition in which to run atsflux jobs.",
+    )
+    return parser.parse_known_args()
+
+
+def main():
+    """
+    Wrapper driver for running ATS tests under Flux.
+
+    Available command-line arguments:
+        --numNodes=, -N
+          Specify a number of nodes to use, will default to use 3.
+        --bank=
+          Specify a project bank to charge, will default to use whatever your default bank is.
+        --partition=
+          Specify either the debug or batch partition, will default to use debug.
+    """
+
+    """Argument parsing logic goes first."""
+    args, extra_args = _parse_args()
+
+    """Check to make sure a valid Flux installation is present on the system."""
+    try:
+        version = int(
+            sp.check_output(["flux", "-V"])
+            .split()[1]
+            .decode("utf-8")
+            .split(".")[1]
+        )
+        if (version) < 38:
+            sys.exit(
+                f"""Error: this system does not have a current version of Flux installed.
+                Version 0.{version}.0 is installed but >= 0.38.0 is required. Please update
+                the Flux version or use a different Flux (check which one you're using with `which flux`).
+                """
+            )
+    except Exception:
+        sys.exit(
+            """Error: this system does not have a current version of Flux installed.
+            No version of Flux was found on this system but version >= 0.38.0 required. Please update
+            the Flux version or specify a different Flux installation to use.
+            """
+        )
+
+    # TODO: 'sys_type' unused.
+    # sys_type = os.getenv("SYS_TYPE")
+    slurm_job_id = os.getenv("SLURM_JOB_ID")
+    cmd = []
+
+    if slurm_job_id == None:  ## if this is on login node
+        cmd = [
+            "salloc",
+            f"--nodes={args.numNodes}",
+            f"--partition={args.partition}",
+            f"--account={args.account}",
+            "--exclusive",
+            "--time=60",
+        ]
+
+    """Get the path to an .ats file and find the test file."""
+    """Find the proper ATS implementation to pass the complete path."""
+    myats = os.path.join(sys.exec_prefix, "bin", "ats")
+
+    """The following is copied from atslite1.pyL#28-46"""
+    test_ats_file = ""
+
+    # TODO: test_ats_file should be found in extra_args
+    # TODO: Behavior of extra_args: will this also pass any ats-specific arguments we might want
+    
+    cmd.extend(
+        [
+            "srun",
+            f"-N{args.numNodes}",
+            f"-n{args.numNodes}",
+            "--pty",
+            "/usr/bin/flux",
```

Comment:
I would replace this with [shutil.which("flux")](https://docs.python.org/3.8/library/shutil.html#shutil.which). By this point you're guaranteed to have Flux in PATH because you ran `flux -V` already, but there's no assurance that it will be in `/usr/bin`.